01-data-structures/07-kevin-bacon/kb_finder.rb

```diff
@@ -0,0 +1,41 @@
+require_relative 'node'
+require_relative 'movie'
+def find_kevin_bacon(node)
+  if node.name == "Kevin Bacon"
+    return []
+  end
+
+  queue = [node]
+  path = {}
+  current_node = node
+  path[current_node.name] = {:distance => 0, 
+                             :previous_node => nil, 
+                             :encountered_movie => nil}
+
+  while (!queue.empty? and path[current_node.name][:distance] < 5)
```

Comment:
I don't think there is any way around this as node.film_actor_hash is a nested {"movie1" : [actor1, actor2, actor3]} where actorX are actor nodes.